File: 83-remove-duplicates-from-sorted-list/remove-duplicates-from-sorted-list.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* deleteDuplicates(struct ListNode* head) {
    if (!head) return NULL;
    struct ListNode* current = head;
    while (head->next) {
        if (head->val == head->next->val) {
            head->next = head->next->next;
        }
        if (head->next && head->val != head->next->val) {
            head = head->next;
        }
    }
    return current;
}

File: 83-remove-duplicates-from-sorted-list/remove-duplicates-from-sorted-list.rs
// Definition for singly-linked list.
// #[derive(PartialEq, Eq, Clone, Debug)]
// pub struct ListNode {
//   pub val: i32,
//   pub next: Option<Box<ListNode>>
// }
// 
// impl ListNode {
//   #[inline]
//   fn new(val: i32) -> Self {
//     ListNode {
//       next: None,
//       val
//     }
//   }
// }
impl Solution {
    pub fn delete_duplicates(head: Option<Box<ListNode>>) -> Option<Box<ListNode>> {
        let mut head = head;
        let mut current = &mut head;
        while let Some(node) = current {
            while let Some(next) = &node.next {
                if node.val == next.val {
                    node.next = next.next.clone();
                } else {
                    break;
                }
            }
            current = &mut node.next;
        }
        head
    }
}